hasher-matcher-actioner/hmalib/lambdas/api/api_auth.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import os
+import json
+import time
+import jwt
+import requests
+import base64
+from jwt.algorithms import RSAAlgorithm
+
+from hmalib.common.logging import get_logger
+
+ACCESS_TOKEN = os.environ["ACCESS_TOKEN"]
+USER_POOL_URL = os.environ["USER_POOL_URL"]
+CLIENT_ID = os.environ["CLIENT_ID"]
+
+keys_url = f"{USER_POOL_URL}/.well-known/jwks.json"
+
+response = requests.get(keys_url)
+key_list = json.loads(response.text).get("keys", [])
+keys = {key["kid"]: json.dumps(key) for key in key_list}
+
+
+def is_jwt(token: str) -> bool:
```

Comment:
nit: this could be done by pyjwt itself, no?

pyjwt.decode throws a DecodeError if it can't process the JWT.

https://pyjwt.readthedocs.io/en/stable/api.html#jwt.exceptions.DecodeError